app/assets/javascripts/settings/ldap-conn-test.js

```diff
@@ -1,117 +1,155 @@
+    return true;
+  }
+
+  function renderValidity(msg, isValid) {
+    $('#ldap_conn_message').removeClass('ldap-conn-message-default ldap-conn-message-valid ldap-conn-message-invalid');
+
     if (isValid) {
-      $('#ldap_conn_save').removeProp('disabled')
-      $('#ldap_conn_message').addClass('ldap-conn-message-valid')
+      $('#ldap_conn_save').removeProp('disabled');
+      $('#ldap_conn_message').addClass('ldap-conn-message-valid');
     } else {
-      $('#ldap_conn_save').prop('disabled', 'disabled')
-      $('#ldap_conn_message').addClass('ldap-conn-message-invalid')
+      $('#ldap_conn_save').prop('disabled', 'disabled');
+      $('#ldap_conn_message').addClass('ldap-conn-message-invalid');
     }
-    $('#ldap_conn_message').html(msg)
+
+    $('#ldap_conn_message').html(msg);
   }
 
-  function ldapConnectionTestPart1 () {
-    var file = $('#dex_connector_ldap_certificate').val()
-    if(!file) {
-      ldapConnectionTestPart2( $('#dex_connector_ldap_current_cert').val() )
-      return;
+  function ldapConnectionTestPart2(cert) {
+    var host = $('#dex_connector_ldap_host').val();
+    var port = $('#dex_connector_ldap_port').val();
+    var startTLS = $('#dex_connector_ldap_start_tls_true').parent().hasClass('active');
+    // Flip logic of #anonymous_bind_toggle since expanded div means anonymous binding is 'false'
+    var anonBind = ($('#anonymous_bind_toggle').attr('aria-expanded') === 'false');
+    var dn = $('#dex_connector_ldap_bind_dn').val();
+    var pass = $("#dex_connector_ldap_bind_pw[required='required']").val();
+    var baseDN = $('#dex_connector_ldap_user_base_dn').val();
+    var filter = $('#dex_connector_ldap_user_filter').val();
+    var paramArray = [host, port, cert, baseDN, filter];
+
+    if (!anonBind) {
+      paramArray.push(dn, pass);
     }
+
+    if (noneEmpty(paramArray)) { // Check if form inputs are filled in
+      $.post('/settings/ldap_test', {
+        host: host,
+        port: port,
+        start_tls: startTLS,
+        cert: cert,
+        anon_bind: anonBind,
+        dn: dn,
+        pass: pass,
+        base_dn: baseDN,
+        filter: filter
+      }).done(function (data) {
+        if (data.result.test_pass) {
+          renderValidity('Sucessfully validated connection to LDAP server', true);
+        } else {
+          renderValidity('Validation failure. ' + data.result.message, false);
+        }
+      }).fail(function (status) {
+        var valMessage = 'Failed to connect to Velum server with following error code/message: ' + status.status + '; ' + status.statusText;
+        renderValidity(valMessage, false);
+      });
+    } else {
+      renderValidity('Missing form data', false);
+    }
+  }
+
+  function ldapConnectionTestPart1() {
+    var file = $('#dex_connector_ldap_certificate').val();
     var fileInput = $('#dex_connector_ldap_certificate')[0];
```

Comment:
As one of the linting rules, variables should always be declared on the top of the scope. So I've just moved both lines to the top of the function to satisfy the linter. We could've declared only the variables and assigned the values after the `if` block but I thought this would be simpler. There's no side effect on the functionality itself.